WordPress/Classes/ViewRelated/NUX/LoginViewController.m

```diff
@@ -240,6 +254,112 @@ - (WPWalkthroughOverlayView *)baseLoginErrorOverlayView:(NSString *)message
+
+- (void)autoFillLoginWithSharedWebCredentialsIfAvailable
+{
+    __weak __typeof(self)weakSelf = self;
+    [self requestSharedWebCredentials:^(NSString *username, NSString *password) {
+        
+        if (!username.length || !password.length) {
+            return;
+        }
+        if (!weakSelf.viewModel.userIsDotCom) {
+            // If the user has swtiched away from dotcom sign-in, swith back before autofilling
+            [weakSelf.viewModel toggleSignInFormAction];
+        }
+        
+        // Update the model
+        weakSelf.viewModel.username = username;
+        weakSelf.viewModel.password = password;
+        // Update the fields for display
+        [weakSelf setUsernameTextValue:username];
+        [weakSelf setPasswordTextValue:password];
+        
+        weakSelf.autofilledUsernameCredentialHash = [username hash];
+        weakSelf.autofilledPasswordCredentialHash = [password hash];
+        
+        [WPAnalytics track:WPAnalyticsStatSafariCredentialsLoginFilled];
+    }];
+}
+
+- (void)updateAutoFillLoginCredentialsIfNeeded:(NSString *)username password:(NSString *)password
+{
+    // Don't try and update credentials for self-hosted.
+    if (!self.viewModel.userIsDotCom) {
+        return;
+    }
+    
+    // If the user changed screen names, don't try and update/create a new shared web credential.
+    // We'll let Safari handle creating newly saved usernames/passwords.
+    if (self.autofilledUsernameCredentialHash != [username hash]) {
+        return;
+    }
+    
+    // If the user didn't change the password from previousl filled password no update is needed.
+    if (self.autofilledPasswordCredentialHash == [password hash]) {
+        return;
+    }
+    
+    // Update the shared credential
+    CFStringRef fqdnStr = (__bridge CFStringRef)LoginSharedWebCredentialFQDN;
+    CFStringRef usernameStr = (__bridge CFStringRef)username;
+    CFStringRef passwordStr = (__bridge CFStringRef)password;
+    SecAddSharedWebCredential(fqdnStr, usernameStr, passwordStr, ^(CFErrorRef  _Nullable error) {
+        if (error) {
+            DDLogError(@"Error occurred updating shared web credential: %@", error);
+            return;
+        }
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [WPAnalytics track:WPAnalyticsStatSafariCredentialsLoginUpdated];
+        });
+    });
+}
+
+- (void)requestSharedWebCredentials:(void(^)(NSString *username, NSString *password))completion
+{
+    if (self.shouldAvoidRequestingSharedCredentials) {
+        return;
+    }
+    
+    // Disable repeat calls for shared credentials.
+    self.shouldAvoidRequestingSharedCredentials = YES;
+    CFStringRef fqdnStr = (__bridge CFStringRef)LoginSharedWebCredentialFQDN;
+    SecRequestSharedWebCredential(fqdnStr, NULL, ^(CFArrayRef credentials, CFErrorRef error) {
+        
+        if (error != NULL) {
+            DDLogError(@"Error occurred while requesting shared web credentials: %@", error);
```

Comment:
I see what you mean, not necessarily an error per-se. What do you think about `Completed requesting shared web credentials with: %@`?